content/blog/webr-0-3-1/index.md

```diff
@@ -0,0 +1,368 @@
+
+You can install the latest release from [npm](https://www.npmjs.com/package/webr) with the command:
+
+    npm i webr@0.3.1
+
+or if you're using JavaScript modules to import webR directly from CDN:
+
+``` javascript
+import { WebR } from 'https://webr.r-wasm.org/v0.3.1/webr.mjs';
+```
+
+A summary of changes is described below, with the full [release notes](https://github.com/r-wasm/webr/releases) on GitHub.
+
+## Evaluating R code from JavaScript
+
+The underlying interpreter powering webR is built from the same source code as R itself, with patches applied so that it can run in the WebAssembly environment. With this release, we have rebased our patches on the latest stable version of R[^1]. By keeping our source in sync, improvements and bug fixes made by the R Core Team also benefit any project making use of webR.
+
+WebR's core functionality is to evaluate R code from a JavaScript environment. As such, it is imperative that this works well, even with large and complex scripts. The [webR app](https://webr.r-wasm.org/v0.3.1/) has been updated to better handle large R scripts, and scripts longer than 4096 characters should no longer cause strange issues in the R console.
+
+### Loading WebAssembly packages
+
+The package management functions provided by webR have been expanded and improved. Once enabled [^2], [`install.packages()`](https://rdrr.io/r/utils/install.packages.html), [`library()`](https://rdrr.io/r/base/library.html) and [`require()`](https://rdrr.io/r/base/library.html) are shimmed so that installing or loading R packages automatically downloads WebAssembly binaries from the [webR package repository](https://repo.r-wasm.org). Also, it is no longer required to run the [`library()`](https://rdrr.io/r/base/library.html) command a second time to subsequently load the package.
+
+In this interactive example, webR is configured to automatically install WebAssembly packages. Click "Run code" to download the packages listed in the R script.
+
+<div class="highlight">
+
+{{< webr-editor code=`# Explicitly install wasm packages\ninstall.packages("cli")\n\n# Automatically install wasm packages\nlibrary(vctrs)\nrequire(jsonlite)\n\n# Confirm the packages installed successfully\nrownames(installed.packages())` width=504 height=311.472 >}}
+
+</div>
+
+See the [documentation](https://docs.r-wasm.org/webr/latest/packages.html) for more details on how to control this behaviour in your own webR-powered applications, including optionally showing an interactive download menu to the user.
+
+### Error handling and reporting
+
+Improvements have been made to how webR raises R conditions as JavaScript exceptions. Exceptions now include the offending source R call in the error message text, better matching what is shown in a traditional R console.
+
+``` javascript
+await webR.evalR("sin('abc')");
+```
+
+<div class="output error">
+
+    Uncaught Error: Error in sin("abc"): non-numeric argument to mathematical function
```

Comment:
Unrelated to the post and not terribly important but could the call be backticked? Less consistent with base R but to be perfectly consistent we'd need a space between the call and semi-colon, I think it's better to follow the tidyverse style here.

Reply:
I agree, I don't think we need it to be entirely consistent with base R and I like the idea of using tidyverse style with backticks. I've made an issue to track this: https://github.com/r-wasm/webr/issues/397